### PR TITLE
Cubed sphere validation experiment

### PR DESCRIPTION
**Pull request by ali-ramadhan**
This PR adds a validation experiment as suggested by @christophernhill and @jm-c to add a very small Gaussian sea surface height η perturbation in the northern hemisphere of the cubed sphere face and watch it evolve.

There is no rotation/Coriolis and no diffusivity right now to test the simplest model possible (so η just splashes around as expected).

Next step would be to turn on rotation, diffusivity, and the implicit free surface. Might have to generalize some things, e.g. Coriolis, to non-regular curvilinear grids like `ConformalCubedSphereFaceGrid`. So I guess the main purpose of this PR is to make sure the cubed sphere face works, so maybe the validation experiment won't make it into the final version of this PR.

https://user-images.githubusercontent.com/20099589/111330144-139fbb00-8646-11eb-9598-a0e18e88280b.mp4

This PR depends on PR #1433. Will `git rebase main` once it's merged.

### CODE REVIEW

**Comment by francispoulin** [comment]
Great looking surface gravity waves you have there!  They are pretty much doing what I would expect and it's great to see.  When you turn on rotation I expect to see geostrophic adjustment and a vortex left in the center.  Very nice.

**Comment by ali-ramadhan** [comment]
Not sure if this is fully correct but qualitatively the Gaussian blob remains coherent due to geostrophic balance and we get Kelvin waves on the boundary.

Was hoping we could merge this PR as it gets the cubed sphere face grid working (and it seems to be working okay pending more rigorous validation tests). Not sure if we'll keep the surface gravity waves validation experiment but might be good to use it for the full cubed sphere as well.

https://user-images.githubusercontent.com/20099589/112829246-d6342800-905e-11eb-89fc-4f77a3eb8521.mp4

File: src/Grids/conformal_cubed_sphere_face_grid.jl [hunk 1]
```diff
@@ -157,8 +157,8 @@ end
 
 function offset_data(data, Hx, Hy)
```
Comment (glwagner) [inline src/Grids/conformal_cubed_sphere_face_grid.jl:158]: There's another `offset_data` function elsewhere right?

**Review by glwagner — APPROVED** [review]

File: src/Grids/conformal_cubed_sphere_face_grid.jl (hunk 1 above)
Comment (ali-ramadhan) [reply src/Grids/conformal_cubed_sphere_face_grid.jl:158]: True. https://github.com/CliMA/Oceananigans.jl/blob/c7c9a3b41452a7f3934ba2903faa7df149c5d794/src/Fields/new_data.jl#L33

Not sure what I wrote this. Will fix in PR #1522.

**Review by christophernhill — APPROVED** [review]
👍

**Pull request merged**